packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/InternalTypedRealm.kt

```diff
@@ -34,4 +38,67 @@ internal interface InternalTypedRealm : TypedRealm {
         val className = configuration.mediator.companionOf(clazz).`io_realm_kotlin_className`
         return ObjectQuery(realmReference, realmReference.schemaMetadata.getOrThrow(className).classKey, clazz, configuration.mediator, null, query, *args)
     }
+
+    private fun <T : BaseRealmObject> copyObjectFromRealm(obj: T, depth: Int, closeAfterCopy: Boolean, cache: ManagedToUnmanagedObjectCache): T {
+        // Be able to inject a cache here as well, so the Iterable case can share the cache
+        if (!obj.isManaged()) {
+            throw IllegalArgumentException("This object is unmanaged. Only managed objects can be copied: $obj.")
+        }
+        if (!obj.isValid()) {
+            throw IllegalArgumentException(
+                "Only valid objects can be copied from Realm. " +
+                    "This object was either deleted or the Realm has been closed, making this " +
+                    "object invalid: $obj."
+            )
+        }
+        if (depth < 0) {
```

Comment:
Shouldn't depth be `1` at least? unless you consider the root object at level `0`

Reply:
The root is considered level 0, this should also be part of the Dokka (although It was a massive wall of text).